whylog/tests/tests_config/test_compare_with_bound.py

```diff
@@ -0,0 +1,44 @@
+from datetime import datetime
+from unittest import TestCase
+
+from whylog.config.investigation_plan import InvestigationStep
+from whylog.config.utils import CompareResult
+
+
+class TestBoundCompare(TestCase):
+    def test_compare_with_bound(self):
```

Comment:
this test is data-driven, but very hard to read. Please make it so that it is easily understandable. By the way I think there are 4 or 7 test cases here.